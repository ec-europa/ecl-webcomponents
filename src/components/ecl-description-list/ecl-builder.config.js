const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  scripts: [
    {
      entry: path.resolve(nodeModules, '@ecl/vanilla-component-description-list/description-list.js'),
      dest: path.resolve(outputFolder, 'scripts/ecl-description-list-vanilla.js'),
      options: {
        banner,
        moduleName: 'ECL',
        includePaths,
        sourceMap: false,
      },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-description-list-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-description-list-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-description-list-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-description-list-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
  ],
};
