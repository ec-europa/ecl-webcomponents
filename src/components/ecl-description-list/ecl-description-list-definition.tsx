import { Component, Prop, h } from '@stencil/core';
import Fragment from 'stencil-fragment'

@Component({
  tag: 'ecl-description-list-definition',
  shadow: false,
})
export class EclDescriptionListDefinition {
  @Prop() type: string = 'text';
  @Prop() styleClass: string = '';
  @Prop() theme: string = 'ec';
  @Prop() items: string;

  getClass(): string {
    return [
      `ecl-description-list__definition`,
      `sc-ecl-description-list-${this.theme}`,
      `ecl-description-list__definition--${this.type}`,
      this.styleClass
    ].join(' ');
  }

  render() {
    const itemsArray = this.items ? JSON.parse(this.items) : '';
    console.log(itemsArray);
    return (
      <dd class={this.getClass()}>
      { this.type === 'link' ?
        <Fragment>
        { itemsArray ? itemsArray.map((link) => (
          <ecl-link
            style-class={`ecl-description-list__definition-item sc-ecl-description-list-${this.theme}`}
            path={link.path}
          >
        { link.icon ? 
          <ecl-icon
            slot="icon-before"
            icon={link.icon}
          >
          </ecl-icon>
          : '' }
            {link.label}
          </ecl-link> 
          )) : <slot></slot>
        }
        </Fragment> : ''
      }

      { this.type === 'inline' ?
        <Fragment>
        { itemsArray ? itemsArray.map((inline) => (
          <ecl-link
            style-class={`ecl-description-list__definition-item sc-ecl-description-list-${this.theme}`}
            path={inline.path}
          >
            {inline.label}
          </ecl-link> 
          )) : <slot></slot>
        } 
        </Fragment> : ''
      }

      { this.type === 'taxonomy' ?
        <Fragment>
        { itemsArray ? itemsArray.map((taxonomy) => (
          <span
            class={`ecl-description-list__definition-item sc-ecl-description-list-${this.theme}`}
          >
            { taxonomy.path ? 
              <ecl-link
                path={taxonomy.path}
              >
                {taxonomy.label}
              </ecl-link> : taxonomy
            }
          </span> 
          )) : <slot></slot>
        }
        </Fragment> : ''
      }

      { this.type === 'text' ? 
        <slot></slot> : '' 
      }
      </dd>
    )
  }
}