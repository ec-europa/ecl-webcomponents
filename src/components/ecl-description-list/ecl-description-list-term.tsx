import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'ecl-description-list-term',
  shadow: false,
  scoped: false,
})
export class EclDescriptionListTerm {
  @Prop() styleClass: string = '';
  @Prop({ mutable: true }) theme: string;

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
  }

  getClass(): string {
    return [
      `ecl-description-list__term`,
      `sc-ecl-description-list-${this.theme}`,
      this.styleClass
    ].join(' ');
  }

  render() {
    return (
      <dt class={this.getClass()}>
        <slot></slot>
      </dt>
    );
  }
}