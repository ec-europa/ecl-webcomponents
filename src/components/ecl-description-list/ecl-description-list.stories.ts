export default {
  title: 'Components/List/description-list',
};

const Template = (args) =>
  `<ecl-description-list
    theme="${args.theme}"
    variant="${args.variant}"
    visible-items="2"
    more-label="See all items"
    ecl-script
  >
    <ecl-description-list-term
      theme="${args.theme}"
    >
      Standard text
    </ecl-description-list-term>
    <ecl-description-list-definition
      type="text"
      theme="${args.theme}"
    >
      Lorem ipsum dolor sit amet, <ecl-link path="/example" title"title">consectetur adipiscing elit</ecl-link>. Suspendisse ut sapien condimentum, aliquet turpis sit amet, finibus purus. Donec porttitor iaculis felis ut dapibus. Sed blandit, massa ac suscipit facilisis
    </ecl-description-list-definition>
    <ecl-description-list-term
      theme="${args.theme}"
    >
      Standalone links
    </ecl-description-list-term>
    <ecl-description-list-definition
      type="link"
      theme="${args.theme}"
      items='[{"label": "Lorem ipsum dolor sit amet", "path": "/example", "icon": "copy"},{"label": "Lorem ipsum dolor sit amet", "path": "/example", "icon": "download"}]'
    >
    </ecl-description-list-definition>
    <ecl-description-list-term
      theme="${args.theme}"
    >
      Links inline
    </ecl-description-list-term>
    <ecl-description-list-definition
      type="inline"
      theme="${args.theme}"
      items='[{"label": "Lorem ipsum dolor sit amet", "path": "/example"},{"label": "Lorem ipsum dolor sit amet", "path": "/example"}]'
    >
    </ecl-description-list-definition>
    <ecl-description-list-term
      theme="${args.theme}"
    >
      Taxonomy list
    </ecl-description-list-term>
    <ecl-description-list-definition
      type="taxonomy"
      theme="${args.theme}"
      collapsible
      items='["Taxonomy item 1", {"label": "Taxonomy item 2", "path": "/example"}, "Taxonomy item 3"]'
    >
    </ecl-description-list-definition>
  </ecl-description-list>`;

export const DescriptionList = Template.bind({});
DescriptionList.storyName = 'default';
DescriptionList.args = {
  variant: 'vertical',
};
DescriptionList.argTypes = {
  variant: {
    type: { name: 'select'},
    options: [
      'vertical',
      'horizontal',
    ],
    description: 'Choose variant',
  },
};