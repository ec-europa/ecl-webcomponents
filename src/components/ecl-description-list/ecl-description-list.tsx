import { Component, Prop, h, Element } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";
declare const ECL: any;

@Component({
  tag: 'ecl-description-list',
  styleUrls: {
    ec: './build/styles/ecl-description-list-ec.css',
    eu: './build/styles/ecl-description-list-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})
export class EclDescriptionList {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() theme: string = 'ec';
  @Prop() variant: string = 'vertical';
  @Prop() visibleItems: number;
  @Prop() moreLabel: string;
  @Prop() eclScript: boolean = false;

  getClass(): string {
    return [
      `ecl-description-list`,
      `ecl-description-list--${this.variant}`,
      this.styleClass
    ].join(' ');
  }

  getAttrs() {
    const attrs = {
      'class': this.getClass(),
    };

    if (this.visibleItems) {
      attrs['data-ecl-description-list'] = true;
      attrs['data-ecl-description-list-more-label'] = this.moreLabel;
      attrs['data-ecl-description-list-visible-items'] = this.visibleItems;
    }

    return attrs;
  }

  componentDidRender() {
    if (this.eclScript) {
      const src = getAssetPath('./build/scripts/ecl-description-list-vanilla.js');
      if (document.querySelector(`script[src="${src}"]`)) {
        document.querySelector(`script[src="${src}"]`).remove();
      }
      const script = document.createElement('script');
      script.src = src;
      script.onload = () => {
        const descriptionList = new ECL.DescriptionList(this.el.firstElementChild);
        descriptionList.init();
        const seeMore = this.el.querySelectorAll('.ecl-description-list__see_more');
        if (seeMore[0]) {
          seeMore.forEach((more) => {
            more.classList.add(`sc-ecl-description-list-${this.theme}`);
          });
        }
      };
      document.body.appendChild(script);
    }
  }

  render() {
    return (
      <dl {...this.getAttrs()}>
        <slot></slot>
      </dl>
    );
  }
}
