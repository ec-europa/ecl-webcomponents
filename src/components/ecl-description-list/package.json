{
  "name": "@ecl/web-components-description-list",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.6.0",
  "description": "ECL Link",
  "main": "ecl-description-list.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build:scripts": "cross-env NODE_ENV=production ecl-builder scripts",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.8.4",
    "@ecl/theme-ec": "3.8.4",
    "@ecl/theme-eu": "3.8.4",
    "@ecl/vanilla-component-description-list": "3.8.4",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
