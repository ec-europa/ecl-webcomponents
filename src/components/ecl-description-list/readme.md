# ecl-description-list



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type     | Default |
| ------------ | ------------- | ----------- | -------- | ------- |
| `styleClass` | `style-class` |             | `string` | `''`    |
| `theme`      | `theme`       |             | `string` | `'ec'`  |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
