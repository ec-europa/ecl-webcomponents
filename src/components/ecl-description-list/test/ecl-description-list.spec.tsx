import { newSpecPage } from '@stencil/core/testing';
import { EclDescriptionList } from '../ecl-description-list';
import { EclDescriptionListTerm } from '../ecl-description-list-term';
import { EclDescriptionListDefinition } from '../ecl-description-list-definition';

describe('ecl-description-list', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclDescriptionList, EclDescriptionListDefinition, EclDescriptionListTerm],
      html: `<ecl-description-list
        theme="ec"
        variant="vertical"
      >
        <ecl-description-list-term
          theme="ec"
        >
          Standard text
        </ecl-description-list-term>
        <ecl-description-list-definition
          type="text"
          theme="ec"
        >
          Lorem ipsum dolor sit amet, <a href="/component-library/example" class="ecl-link">consectetur adipiscing elit</a>. Suspendisse ut sapien condimentum, aliquet turpis sit amet, finibus purus. Donec porttitor iaculis felis ut dapibus. Sed blandit, massa ac suscipit facilisis
        </ecl-description-list-definition>
        <ecl-description-list-term
          theme="ec"
        >
          Standalone links
        </ecl-description-list-term>
        <ecl-description-list-definition
          type="link"
          theme="ec"
          items='[{"label": "Lorem ipsum dolor sit amet", "path": "/example.html", "icon": "copy"},{"label": "Lorem ipsum dolor sit amet", "path": "/example.html", "icon": "download"}]'
        >
        </ecl-description-list-definition>
        <ecl-description-list-term
          theme="ec"
        >
          Links inline
        </ecl-description-list-term>
        <ecl-description-list-definition
          type="inline"
          theme="ec"
          items='[{"label": "Lorem ipsum dolor sit amet", "path": "/example.html"},{"label": "Lorem ipsum dolor sit amet", "path": "/example.html"}]'
        >
        </ecl-description-list-definition>
        <ecl-description-list-term
          theme="ec"
        >
          Taxonomy list
        </ecl-description-list-term>
        <ecl-description-list-definition
          type="taxonomy"
          theme="ec"
          items='["Taxonomy item 1", {"label": "Taxonomy item 2", "path": "/example.html"}, "Taxonomy item 3"]'
        >
        </ecl-description-list-definition>
      </ecl-description-list>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
