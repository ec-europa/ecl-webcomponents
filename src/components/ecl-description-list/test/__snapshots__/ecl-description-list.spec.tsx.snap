// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-description-list renders 1`] = `
<ecl-description-list theme="ec" variant="vertical">
  <!---->
  <dl class="ecl-description-list ecl-description-list--vertical">
    <ecl-description-list-term theme="ec">
      Standard text
    </ecl-description-list-term>
    <ecl-description-list-definition theme="ec" type="text">
      Lorem ipsum dolor sit amet,
      <a class="ecl-link" href="/component-library/example">
        consectetur adipiscing elit
      </a>
      . Suspendisse ut sapien condimentum, aliquet turpis sit amet, finibus purus. Donec porttitor iaculis felis ut dapibus. Sed blandit, massa ac suscipit facilisis
    </ecl-description-list-definition>
    <ecl-description-list-term theme="ec">
      Standalone links
    </ecl-description-list-term>
    <ecl-description-list-definition items="[{&quot;label&quot;: &quot;Lorem ipsum dolor sit amet&quot;, &quot;path&quot;: &quot;/example.html&quot;, &quot;icon&quot;: &quot;copy&quot;},{&quot;label&quot;: &quot;Lorem ipsum dolor sit amet&quot;, &quot;path&quot;: &quot;/example.html&quot;, &quot;icon&quot;: &quot;download&quot;}]" theme="ec" type="link"></ecl-description-list-definition>
    <ecl-description-list-term theme="ec">
      Links inline
    </ecl-description-list-term>
    <ecl-description-list-definition items="[{&quot;label&quot;: &quot;Lorem ipsum dolor sit amet&quot;, &quot;path&quot;: &quot;/example.html&quot;},{&quot;label&quot;: &quot;Lorem ipsum dolor sit amet&quot;, &quot;path&quot;: &quot;/example.html&quot;}]" theme="ec" type="inline"></ecl-description-list-definition>
    <ecl-description-list-term theme="ec">
      Taxonomy list
    </ecl-description-list-term>
    <ecl-description-list-definition items="[&quot;Taxonomy item 1&quot;,{&quot;label&quot;: &quot;Taxonomy item 2&quot;, &quot;path&quot;: &quot;/example.html&quot;},&quot;Taxonomy item 3&quot;]" theme="ec" type="taxonomy"></ecl-description-list-definition>
  </dl>
</ecl-description-list>
`;
